sim/xcelium/__init__.py

```diff
@@ -0,0 +1,394 @@
+    self.write_header(header, f)    
+    
+    f.write("\n# XRUN OPTIONS: \n")
+    [f.write(elem + "\n") for elem in xrun_opts_proc.values() if elem is not None]
+    f.write("\n# SIM OPTIONS: \n")
+    [f.write(elem + "\n") for elem in sim_opts_proc.values() if elem is not None]
+    for opt_list in additional_opt: 
+      if opt_list[1]: 
+        f.write(f"\n# {opt_list[0]} OPTIONS: \n")
+        [f.write(elem + "\n") for elem in opt_list[1]]
+    f.close()  
+    
+    return arg_path  
+  
+  # Deposit values
+  # Try to maintain some parity with vcs plugin.
+  def generate_gl_deposit_tcl(self) -> List[str]:
+    sim_opts  = self.extract_sim_opts() [1]
+    tb_prefix = sim_opts["tb_name"] + '.' + sim_opts["tb_dut"]
+    force_val = sim_opts["gl_register_force_value"]
+    
+    abspath_all_regs = os.path.join(os.getcwd(), self.all_regs)
+    if not os.path.isfile(abspath_all_regs):
+      self.logger.error("List of all regs json not found as expected at {0}".format(self.all_regs))
+
+    formatted_deposit = []
+    with open(abspath_all_regs) as reg_file:
+      reg_json = json.load(reg_file)
+      assert isinstance(reg_json, List), "list of all sequential cells should be a json list of dictionaries from string to string not {}".format(type(reg_json))
+      for reg in sorted(reg_json, key=lambda r: len(r["path"])): 
+        path = reg["path"]
+        path = path.split('/')
+        special_char =['[',']','#','$',';','!',"{",'}','\\']
+        path = ['@{' + subpath + ' }' if any(char in subpath for char in special_char) else subpath for subpath in path]
+        path='.'.join(path)
+        pin = reg["pin"]
+        formatted_deposit.append("deposit " + tb_prefix + "." + path + "." + pin + " = " + str(force_val))
+        
+    return formatted_deposit
+
+  # Creates an sdf cmd file for command line driven sdf annotation.
+  # Until sdf annotation provides values other than maximum, sdf_cmd_file will only support mtm max.
+  def generate_sdf_cmd_file(self) -> bool:
+    sim_opts  = self.extract_sim_opts()[1]
+    prefix = sim_opts["tb_name"] + '.' + sim_opts["tb_dut"]
+
+    f = open(self.sdf_cmd_file,"w+")
+    f.write(f'SDF_FILE = "{self.sdf_file}", \n')
+    f.write(f'MTM_CONTROL = "MAXIMUM", \n')
+    f.write(f'SCALE_TYPE = "FROM_MAXIMUM", \n')
+    f.write(f'SCOPE = {prefix};')
+    f.close()
+    return True
+
+  # Creates saif arguments for tcl commands for tcl driver.
+  def generate_saif_tcl_cmd(self) -> str:
+    saif_opts = self.extract_saif_opts()
+    sim_opts  = self.extract_sim_opts()[1]
+    prefix = sim_opts["tb_name"] + '.' + sim_opts["tb_dut"]
+
+    saif_args = ""
+
+    # Process saif options
+    if saif_opts["mode"] == "time":
+      saif_start_time = saif_opts["start_time"]
+      saif_end_time   = saif_opts["end_time"]
+    elif saif_opts["mode"] == "full":
+      pass
+    else:
+      self.logger.warning("Bad saif_mode:${saif_mode}. Valid modes are time, full, or none. Defaulting to none.")
```

Comment:
Are you able to support any of the `trigger` saif modes?

Reply:
No support for trigger key but can support raw_trigger. Added!